src/Symfony/Component/Yaml/Tests/ParserTest.php

```diff
@@ -66,14 +66,28 @@ public function testTabsInYaml()
             try {
                 $this->parser->parse($yaml);
 
-                $this->fail('YAML files must not contain tabs');
+                $this->fail('YAML files must not contain tabs as indentation');
             } catch (\Exception $e) {
-                $this->assertInstanceOf(\Exception::class, $e, 'YAML files must not contain tabs');
-                $this->assertEquals('A YAML file cannot contain tabs as indentation at line 2 (near "'.strpbrk($yaml, "\t").'").', $e->getMessage(), 'YAML files must not contain tabs');
+                $this->assertInstanceOf(\Exception::class, $e, 'YAML files must not contain tabs as indentation');
+                $this->assertEquals('A YAML file cannot contain tabs as indentation at line 2 (near "'.strpbrk($yaml, "\t").'").', $e->getMessage(), 'YAML files must not contain tabs as indentation');
             }
         }
     }
 
+    public function testTabsAsSeparationBetweenTokensInYaml()
+    {
+        // test tabs as separation between tokens
+        $yamls = [
+            'foo:	bar', // tab after colon
+            'foo: 	bar', // space and tab after colon
+        ];
+
+        foreach ($yamls as $yaml) {
```

Comment:
this would be better implemented as 2 separate tests IMO

Reply:
I'm open to splitting it up, but I based it on the existing test for tabs as indentation. Should I also change that one then for consistency?

And would a dataprovider also be okay, or do you prefer separate tests?

Reply:
a data provider is fine

Reply:
tests of the YAML component are quite old (that component actually comes from symfony 1 originally), and so they don't really follow PHPUnit best practices. But ideally, new tests should follow them

Reply:
Alright, thanks for clarifying! I haven't had much time to change it so far but I still intend to follow up on this somewhere in the coming days.

Reply:
Out of curiosity, why should tests not have a `: void` return type?